validator/src/main/java/com/amazon/aoc/validators/CWMetricValidator.java

```diff
@@ -73,6 +73,11 @@ public void validate() throws Exception {
     Set<String> skippedDimensionNameList = new HashSet<>();
     for (Metric metric : expectedMetricList) {
       for (Dimension dimension : metric.getDimensions()) {
+
+        if (dimension.getValue() == null || dimension.getValue().equals("")) {
```

Comment:
Might want to look into https://commons.apache.org/ otherwise this is fine with me

Reply:
Looked through it. The equals will have problem [when the reference object is null](https://stackoverflow.com/questions/4501061/java-null-check-why-use-instead-of-equals/4501084#4501084) so I cannot use that. Therefore, it makes more sense for me to use [Object.equals ](https://stackoverflow.com/a/44799807) instead and I have already tested it out without any issue.

Reply:
Seems it is not feasible to use Object.equals since the PR build shows the errors https://github.com/aws-observability/aws-otel-test-framework/runs/4539943758?check_suite_focus=true (Cannot compare null with string )